cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -818,7 +824,8 @@ resolveWithoutDependencies (DepResolverParams targets constraints
                            . InstalledPackageIndex.lookupSourcePackageId
                                                      installedPkgIndex
                            . packageId
-        versionPref   pkg = packageVersion pkg `withinRange` preferredVersions
+        versionPref   pkg = packageVersion pkg `withinRange`
+                            combinePrefRanges preferredVersions
```

Comment:
Why does this function handle preferences differently than in the modular solver?  This seems more consistent:
```versionPref   pkg = length $ filter (packageVersion pkg `withinRange`) preferredVersions```